.circleci/config.yml

```diff
@@ -148,10 +148,12 @@ e2e-dind-test: &e2e-dind-test
         path: /tmp/cluster_state
     - run:
         name: Bring down the cluster
+        when: always
```

Comment:
I don't like this for "rerun with ssh" usage of circle for debugging.  I don't think there's any harm in leaving the KDC clusters up since they're on the local workflow's remote_docker instance, which get cleaned up by circleci on job termination, AFAIK.

Reply:
@tiswanso OK, I'll remove that. I just found that surprising by reading the list of steps. How about moving the `store_artifacts` step to the end of the list, like it's done in all istio/istio jobs, for consistency?

Reply:
Actually, I'll just remove the cleanup steps. istio/istio integration tests don't clean up either.